core/src/main/scala/caliban/validation/Validator.scala

```diff
@@ -563,6 +564,60 @@ object Validator {
         )
     }
 
+  private def validateInterface(t: __Type): IO[ValidationError, Unit] = {
+    def validateName(name: String) = {
+      if (name.isEmpty) {
```

Comment:
You could use `IO.whenCase` to make this even shorter and avoid the `else IO.unit`

Reply:
this got replaced by reusing existing code of #244